### PR TITLE
Update urlbar state when the popup selection changes

### LINKED ISSUE #46: correctly update gURLBar state when recommendation is highlighted
When the recommendation gets highlighted, we need to put the URL in the urlbar.

However, if the user keys to an item in the popup, then hits Escape to dismiss the popup, the urlbar state has to be reverted. If we set state in the urlbar correctly, this behavior should be preserved.

There's a buggy implementation of this in the old addon; need to look carefully at `urlbarBindings.xml`, and figure out which of these bits of state is relevant:
- `gURLBar.value`
- `gURLBar._value`
- `gURLBar.inputField.value`
- `gURLBar.valueIsTyped`
- `gURLBar.valueIsPasted`
- `gBrowser.userTypedValue`
- `gBrowser.userTypedClear`

### LINKED ISSUE #47: implement urlbar navigate method
This should be an easy one: if the user wants to navigate to the recommendation URL, we need a method on the urlbar that navigates to the urlbar contents.

I think `gURLBar.handleCommand(url)` should do it, but there are no doubt some funny edge cases.

### PR DESCRIPTION
**Pull request by jaredhirsch**
@chuckharmston R?

Thankfully, the gURLBar.textValue setter lets us hook into an enormously
complex system with a simple one-liner.

Fixes #46, fixes #47.

### CODE REVIEW

File: lib/ui/highlight-manager.js [hunk 1]
```diff
@@ -252,16 +257,35 @@ HighlightManager.prototype = {
...
+      newIndex = evt.forward ? selectedIndex + 1 : selectedIndex - 1;
+      this.popup.el.selectedIndex = newIndex;
+    }
+
+    // Finally, figure out the new url, and tell the urlbar that the selection
+    // has changed.
+    let newUrl;
+    if (newIndex === -1) {
+      let urlEl = this.recommendation.el.querySelector('#universal-search-recommendation-url');
+      newUrl = urlEl && urlEl.textContent;
```
Comment (chuckharmston) [inline lib/ui/highlight-manager.js]: Rather than use `textContent`, can we use a `data-` attribute for meaningful data?

**Comment by chuckharmston** [comment]
This is great work. Feel free to merge at will!

File: lib/ui/highlight-manager.js (hunk 1 above)
Comment (jaredhirsch) [inline lib/ui/highlight-manager.js]: :+1:

**Comment by jaredhirsch** [comment]
Thanks for taking a look! Merging

**Pull request merged**